<a href="https://www.kaggle.com/code/faryalrifaz3374/01-logistic-regression-diabetes-classification?scriptVersionId=253455865" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

#  Step 1: Import libraries

In [1]:
 
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, classification_report,confusion_matrix
from sklearn.linear_model import LogisticRegression

# Step 2: import dataset

In [2]:

data = pd.read_csv('/kaggle/input/diabetes-dataset/diabetes.csv')
print(data.head())

   Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1  


# Step 3: Split the data into features

In [3]:

X = data.drop('Outcome', axis=1)
y = data['Outcome']

# Step 4: Split the data into training and testing sets

In [4]:

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

# Step 5: Create a logistic regression model

In [5]:

model = LogisticRegression()
# train the model
model.fit(X_train, y_train)
# make predictions
y_pred = model.predict(X_test)




/usr/local/lib/python3.10/dist-packages/sklearn/linear_model/_logistic.py:458: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


# Step 6: Evaluate the model

In [6]:

accuracy = accuracy_score(y_test, y_pred)
conf_matrix = confusion_matrix(y_test, y_pred)
report = classification_report(y_test, y_pred)

# Step 7: Print the evaluation metrics

In [7]:

print("Accuracy:", accuracy)
print("Confusion Matrix:\n", conf_matrix)
print("Classification Report:\n", report)

Accuracy: 0.7402597402597403
Confusion Matrix:
 [[121  30]
 [ 30  50]]
Classification Report:
               precision    recall  f1-score   support

           0       0.80      0.80      0.80       151
           1       0.62      0.62      0.62        80

    accuracy                           0.74       231
   macro avg       0.71      0.71      0.71       231
weighted avg       0.74      0.74      0.74       231



# Step 8: Conclusion

The classification results show an overall accuracy of **74%**, indicating that the model correctly predicted the class for most instances. Here’s a summary of key findings:

- **Class 0** (majority class) is predicted with a high precision and recall (both 0.80), meaning the model is effective at identifying true negatives and correctly labeling them.
- **Class 1** (minority class) has lower precision and recall (both 0.62), suggesting the model misses some actual positives and occasionally misclassifies negatives as positives.
- The **confusion matrix** highlights this imbalance, with more errors occurring in predicting Class 1.
- **Macro and weighted averages** (both around 0.71–0.74) further confirm that the model’s performance is stronger on the majority class.

**Key Point:**  
While the model demonstrates solid overall performance, it could be improved further for minority class predictions. This suggests a need for strategies such as rebalancing the dataset, tuning the model, or exploring alternative algorithms to enhance performance, especially for Class 1.